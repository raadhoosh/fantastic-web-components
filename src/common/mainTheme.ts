import { Theme } from './styled';
import * as Color from 'color';
function darken(hex: string) {
    return Color(hex)
        .darken(0.5)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#0066b1',
        secondary_dark: darken('#0066b1'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#d4edda',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    bottom: {
        fontsize: '1rem',
        border: '5px',
    },
    darken: (_color?: string) => {
        return Color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return Color(hex).fade(a ? a : 0.5).string();
    },
};

export default mainTheme;
