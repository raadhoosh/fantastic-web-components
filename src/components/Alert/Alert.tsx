import styled from '../../common/styled';
import Variables from './../Variables';
import * as React from 'react';

/**
 * @description it is props for button component
 */
interface IProps extends Variables.Color {
    backgroundColor?: string;
    color?: string;
    margin?: string;
    hide?: boolean;
    hidden?: boolean;
    children?: any;
    theme?: any
}

const EmAlert = styled('div')(
    (props: (IProps)) => {
        if (props.hide || props.hidden) {
            return { display: 'none' };
        }
        const base_alert = {
            'padding': '.75rem 1.25rem',
            'border':  props.theme.alert.border,
            'border-radius':  props.theme.alert.radius,
            'margin': props.margin ? props.margin : 0,
            'color': props.theme.alert.color,
            'backgroundColor': 'transparent',
            '.hr': {
                'border-top': '1px solid',
                borderTopColor: 'transparent',
            },
        };
        if (props.primary) {
            base_alert.backgroundColor = props.theme.color.primary;
            base_alert.color = props.color ? props.color : props.theme.color.primary_dark;
            base_alert['.hr'].borderTopColor = props.theme.color.primary_dark;
            return base_alert;
        }
        if (props.secondary) {
            base_alert.backgroundColor = props.theme.color.secondary;
            base_alert.color = props.color ? props.color : props.theme.color.secondary_dark;
            base_alert['.hr'].borderTopColor = props.theme.color.secondary_dark;
            return base_alert;
        }
        if (props.success) {
            base_alert.backgroundColor = props.theme.color.success;
            base_alert.color = props.color ? props.color : props.theme.color.success_dark;
            base_alert['.hr'].borderTopColor = props.theme.color.success_dark;
            return base_alert;
        }
        if (props.info) {
            base_alert.backgroundColor = props.theme.color.info;
            base_alert.color = props.color ? props.color : props.theme.color.info_dark;
            base_alert['.hr'].borderTopColor = props.theme.color.info_dark;
            return base_alert;
        }
        if (props.danger) {
            base_alert.backgroundColor = props.theme.color.danger;
            base_alert.color = props.color ? props.color : props.theme.color.danger_dark;
            base_alert['.hr'].borderTopColor = props.theme.color.primary_dark;
            return base_alert;
        }
        if (props.warning) {
            base_alert.backgroundColor = props.theme.color.warning;
            base_alert.color = props.color ? props.color : props.theme.color.warning_dark;
            return base_alert;
        }
        if (props.light) {
            base_alert.backgroundColor = props.theme.color.light;
            base_alert.color = props.color ? props.color : props.theme.color.light_dark;
            base_alert['.hr'].borderTopColor = props.theme.color.light_dark;
            return base_alert;
        }
        if (props.dark) {
            base_alert.backgroundColor = props.theme.color.dark;
            base_alert.color = props.color ? props.color : '#fff';
            base_alert['.hr'].borderTopColor = '#fff';
            return base_alert;
        }

        base_alert.backgroundColor = props.backgroundColor ? props.backgroundColor : props.theme.color.primary;
        base_alert.color = props.color ? props.color : base_alert.color;
        base_alert['.hr'].borderTopColor = props.theme.color.primary_dark;
        return base_alert;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const Alert = (props: IProps) => {

    return (
        <EmAlert {...props}> {props.children} </EmAlert>
    );
};

export default Alert;
