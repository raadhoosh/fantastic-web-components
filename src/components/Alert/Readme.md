React component example:

```js
;<Alert >You can add a custom props to an example wrapper:</Alert>
```

You can add a custom props to an example wrapper:

```js { "props": { "danger": "true" } }
;<Alert danger>You can add a custom props to an example wrapper:!</Alert>

```

Or disable an editor by passing a `warning` modifier:

```jsx noeditor
;<Alert warning>noShadow</Alert>
```
