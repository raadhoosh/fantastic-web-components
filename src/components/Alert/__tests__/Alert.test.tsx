import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import Alert from '../Alert';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const tree = renderer.create(<Alert theme={mainTheme} primary>hi test</Alert>).toJSON();
    expect(tree).toHaveStyleRule('background-color', mainTheme.color.primary);
    expect(tree).toHaveStyleRule('color', mainTheme.color.primary_dark);
    expect(tree).toMatchSnapshot();
});
