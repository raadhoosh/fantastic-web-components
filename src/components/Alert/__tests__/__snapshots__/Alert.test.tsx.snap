// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<div
  className="css-1n132u8"
>
   
  hi test
   
</div>
`;
